### DISCUSSION TITLE
HA Autodetection failure

### DISCUSSION THREAD

**Discussion by baobab67** [allangood/rtlamr2mqtt#75] [Q&A] [answered]
Hi,

I have setup the Rpi  and am successfully sending messages to Mosquito on HA. I have one electricity meter and two water meters. The water meters are automatically discovered by HA, but no success with electricity meter.

I get the error:  
Exception in async_discover when dispatching 'mqtt_discovery_new_sensor_mqtt': ({'name': 'electricity_net_meter', 'unique_id': '30826115', 'unit_of_measurement': 'kWh', 'icon': 'mdi:gauge', 'availability_topic': 'rtlamr/status', 'state_class': 'total_increasing', 'state_topic': 'rtlamr/30826115/state', 'value_template': '{{ value_json.Message.Consumption | float }}', 'json_attributes_topic': 'rtlamr/30826115/state', 'json_attributes_template': '{{ value_json.Message | tojson }}', 'device_class': ['energy'], 'platform': 'mqtt'},) Traceback (most recent call last): File "/usr/src/homeassistant/homeassistant/components/mqtt/mixins.py", line 210, in async_discover config = schema(discovery_payload) File "/usr/local/lib/python3.9/site-packages/voluptuous/validators.py", line 218, in __call__ return self._exec((Schema(val) for val in self.validators), v) File "/usr/local/lib/python3.9/site-packages/voluptuous/validators.py", line 341, in _exec raise e if self.msg is None else AllInvalid(self.msg, path=path) File "/usr/local/lib/python3.9/site-packages/voluptuous/validators.py", line 337, in _exec v = func(v) File "/usr/local/lib/python3.9/site-packages/voluptuous/schema_builder.py", line 272, in __call__ return self._compiled([], data) File "/usr/local/lib/python3.9/site-packages/voluptuous/schema_builder.py", line 817, in validate_callable return schema(data) File "/usr/local/lib/python3.9/site-packages/voluptuous/schema_builder.py", line 272, in __call__ return self._compiled([], data) File "/usr/local/lib/python3.9/site-packages/voluptuous/schema_builder.py", line 594, in validate_dict return base_validate(path, iteritems(data), out) File "/usr/local/lib/python3.9/site-packages/voluptuous/schema_builder.py", line 432, in validate_mapping raise er.MultipleInvalid(errors) voluptuous.error.MultipleInvalid: expected SensorDeviceClass for dictionary value @ data['device_class']

This is the payload: 
{"name": "electricity_net_meter", "unique_id": "30826115", "unit_of_measurement": "kWh", "icon": "mdi:gauge", "availability_topic": "rtlamr/status", "state_class": "total_increasing", "state_topic": "rtlamr/30826115/state", "value_template": "{{ value_json.Message.Consumption | float }}", "json_attributes_topic": "rtlamr/30826115/state", "json_attributes_template": "{{ value_json.Message | tojson }}", "device_class": ["energy"]}

Here is a snippet from my rtlamr2mqtt.yaml file:
meters:
  - id: 30826115
    protocol: scm
    name: electricity_net_meter
    format: "######.###"
    unit_of_measurement: "kWh"
    icon: mdi:gauge
    device_class: energy
  - id: 1480541853
    protocol: r900
    name: water_meter_1
    format: "######.###"
    unit_of_measurement: "\u33A5"
    icon: mdi:gauge
    device_class: none
  - id: 1480541852
    protocol: r900
    name: water_meter_2
    format: "######.###"
    unit_of_measurement: "\u33A5"
    icon: mdi:gauge
    device_class: none

Any suggestions would be gratefully appreciated.

**Comment by allangood** [comment, marked as answer]
This error is caused by a typo and has been fixed in the latest PR. Sorry about that.
Please, wait a few minutes and try to pull the latest version with the fix.

Thank you

**Reply by btehan** [reply]
I believe I had this same issue and 1.5.7 appears to have fixed it.  I just started messing with this yesterday though so I'm still a neophyte.
